File: backend/models/table.py
"""
ERD Viewer - Table & Column Pydantic Models
"""

from typing import Optional
from pydantic import BaseModel


class ColumnDetail(BaseModel):
    name: str
    type_name: str
    nullable: bool = True
    comment: Optional[str] = None
    is_primary_key: bool = False
    is_foreign_key: bool = False


class TableSummary(BaseModel):
    name: str
    full_name: str
    table_type: str  # MANAGED, EXTERNAL, VIEW
    comment: Optional[str] = None
    owner: Optional[str] = None


class TableDetail(TableSummary):
    columns: list[ColumnDetail] = []
    partition_columns: list[str] = []
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    properties: dict[str, str] = {}

File: frontend/src/types/table.ts
export interface ColumnDetail {
  name: string
  type_name: string
  nullable: boolean
  comment: string | null
  is_primary_key: boolean
  is_foreign_key: boolean
}

export interface TableSummary {
  name: string
  full_name: string
  table_type: string
  comment: string | null
  owner: string | null
}

export interface TableDetail extends TableSummary {
  columns: ColumnDetail[]
  partition_columns: string[]
  created_at: string | null
  updated_at: string | null
  properties: Record<string, string>
}
